### PR TITLE
Home work 2

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

File: Homework_2/Program.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            Read = 1,
+            Write,
+            FibonacciNumbers,
+            Exit,
+        }
+        
+        static void Main(string[] args)
+        {
+            IUserAction readAction = new ReadUserAction();
+            int readActionInt = (int) UserActions.Read;
```
Comment (SolovyovDmitry) [inline Homework_2/Program.cs:18]: no space after casting needed

File: Homework_2/Program.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿using Homework_2.UserActions;
+
+namespace Homework_2
+{
+    internal class Program
+    {
+        enum UserActions
```
Comment (SolovyovDmitry) [inline Homework_2/Program.cs]: enums are usually stored in separate file

File: Homework_2/Program.cs (hunk 1 above)
Comment (SolovyovDmitry) [inline Homework_2/Program.cs]: no space after casting required

File: Homework_2/Program.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+                        if (lowerInputKey == "y")
+                        {
+                            userActionToImplementation[UserActions.Exit].MakeAction();
+                        }
+
+                        if (lowerInputKey == "n")
+                        {
+                            break;
+                        }
```
Comment (SolovyovDmitry) [inline Homework_2/Program.cs]: ConsoleKey may be used

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+{
+    public void MakeAction()
+    {
+        while (true)
+        {
+            Console.ForegroundColor = ConsoleColor.Blue;
+            Console.Write("Введите число Фибоначчи: ");
+            Console.ForegroundColor = ConsoleColor.White;
+        
+            if (int.TryParse(Console.ReadLine(), out int fibbonachiNumber) && fibbonachiNumber >= 0)
```
Comment (SolovyovDmitry) [inline Homework_2/UserActions/FibonacciNumbersUserAction.cs]: be consistent. too many different Fibonacci names)

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                    if (lowerInputKey == "n")
+                    {
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    private int CalculateFibonachi(int number)
```
Comment (SolovyovDmitry) [inline Homework_2/UserActions/FibonacciNumbersUserAction.cs]: there is only one usage of this function and it's not meant to be reused. Consider using comment to highlight calculation of Fibonacci numbers

**Review by SolovyovDmitry — COMMENTED** [review]
Good job overall. Don't forget about whitespaces. Your .gitignore file is incorrect

File: Homework_2/UserActions/WriteUserAction.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    private string directoryName = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
+    public void MakeAction()
+    {
+        while (true)
+        {
+            Console.ForegroundColor = ConsoleColor.Blue;
+            Console.Write("Введите URL веб-страницы: ");
+
+            string webURL = Console.ReadLine();
+            string htmlContent = String.Empty;
```
Comment (DecreasedPower) [inline Homework_2/UserActions/WriteUserAction.cs]: use string instead of String

File: Homework_2/UserActions/WriteUserAction.cs [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            
+            try
+            {
+                htmlContent = new WebClient().DownloadString(webURL);
```
Comment (DecreasedPower) [inline Homework_2/UserActions/WriteUserAction.cs]: WebClient is obsolete. Use HttpClient(optional)

Comment (Glunky) [reply Homework_2/UserActions/WriteUserAction.cs]: ++

File: Homework_2/UserActions/WriteUserAction.cs (hunk 6 above)
Comment (Glunky) [reply Homework_2/UserActions/WriteUserAction.cs]: ++

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs (hunk 5 above)
Comment (Glunky) [reply Homework_2/UserActions/FibonacciNumbersUserAction.cs]: ++

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs (hunk 4 above)
Comment (Glunky) [reply Homework_2/UserActions/FibonacciNumbersUserAction.cs]: ++

File: Homework_2/Program.cs (hunk 3 above)
Comment (Glunky) [reply Homework_2/Program.cs]: ++

File: Homework_2/Program.cs (hunk 1 above)
Comment (Glunky) [reply Homework_2/Program.cs]: ++

File: Homework_2/Program.cs (hunk 2 above)
Comment (Glunky) [reply Homework_2/Program.cs]: ++

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    {
+        while (true)
+        {
+            Console.ForegroundColor = ConsoleColor.Blue;
+            Console.Write("Введите число Фибоначчи: ");
+            Console.ForegroundColor = ConsoleColor.White;
+            
+            if (int.TryParse(Console.ReadLine(), out int n) && n >= 0)
+            {
+
```
Comment (TIove) [inline Homework_2/UserActions/FibonacciNumbersUserAction.cs]: Extra empty line

File: Homework_2/Program.cs [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            IUserAction writeAction = new WriteUserAction();
+            int writeActionInt = (int)Constants.UserActions.Write;
+            string writeActionText = Constants.UserActions.Write.ToString();
+            
+            IUserAction fibonacciNumbersAction = new FibonacciNumbersUserAction();
+            int fibonacciNumbersActionInt = (int)Constants.UserActions.FibonacciNumbers;
+            string fibonacciNumbersActionText = Constants.UserActions.FibonacciNumbers.ToString();
+            
+            IUserAction exitAction = new ExitUserAction();
+            int exitActionInt = (int)Constants.UserActions.Exit;
```
Comment (TIove) [inline Homework_2/Program.cs]: Why do you need these variables? just write `(int)Constants.UserActions.Exit` or `Constants.UserActions.Exit.ToString()` whereever you need.

File: Homework_2/UserActions/ReadUserAction.cs [hunk 10]
```diff
@@ -0,0 +1,94 @@
+namespace Homework_2.UserActions;
+
+public class ReadUserAction : IUserAction
+{
+    private string directoryName = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
+    public async Task MakeAction()
```
Comment (TIove) [inline Homework_2/UserActions/ReadUserAction.cs:7]: Empty line is missed

File: Homework_2/UserActions/ReadUserAction.cs [hunk 11]
```diff
@@ -0,0 +1,94 @@
+namespace Homework_2.UserActions;
+
+public class ReadUserAction : IUserAction
+{
+    private string directoryName = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
```
Comment (TIove) [inline Homework_2/UserActions/ReadUserAction.cs:5]: Get file path from the console (optional)

File: Homework_2/UserActions/ReadUserAction.cs [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+                    if (inputKey == ConsoleKey.N)
+                    {
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    private async Task  WriteFileContentToConsole(string filePath, int linesToRead)
```
Comment (TIove) [inline Homework_2/UserActions/ReadUserAction.cs]: An extra space after `Task`

File: Homework_2/UserActions/WriteUserAction.cs [hunk 13]
```diff
@@ -0,0 +1,82 @@
+using System.Net;
+
+namespace Homework_2.UserActions;
+
+public class WriteUserAction : IUserAction
+{
+    private string directoryName = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
```
Comment (TIove) [inline Homework_2/UserActions/WriteUserAction.cs:7]: Get file path from the console (optional)

File: Homework_2/UserActions/FibonacciNumbersUserAction.cs (hunk 8 above)
Comment (Glunky) [reply Homework_2/UserActions/FibonacciNumbersUserAction.cs]: ++

File: Homework_2/Program.cs (hunk 9 above)
Comment (Glunky) [reply Homework_2/Program.cs]: ++

File: Homework_2/UserActions/ReadUserAction.cs (hunk 11 above)
Comment (Glunky) [reply Homework_2/UserActions/ReadUserAction.cs:5]: ++

File: Homework_2/UserActions/ReadUserAction.cs (hunk 12 above)
Comment (Glunky) [reply Homework_2/UserActions/ReadUserAction.cs]: ++

File: Homework_2/UserActions/ReadUserAction.cs (hunk 10 above)
Comment (Glunky) [reply Homework_2/UserActions/ReadUserAction.cs:7]: ++

**Pull request merged**